knowledge-base/validation-custom-dataannotations-validator.md

```diff
@@ -0,0 +1,219 @@
+---
+title: Use Custom DataAnnotations Validator
+description: Learn how to implement and integrate custom DataAnnotations validation with Telerik Blazor components such as Form, Grid, ValidationMessage, ValidationTooltip, and others.
+type: how-to
+page_title: How to Use Custom DataAnnotations Validator with Telerik UI for Blazor
+slug: validation-kb-custom-dataannotations-validator
+tags: telerik, blazor, validation, form, grid
+ticketid: 1666005, 1665269, 1658101, 1560189, 1558247, 1543336
+res_type: kb
+---
+
+## Environment
+
+<table>
+    <tbody>
+        <tr>
+            <td>Product</td>
+            <td>
+                UI for Blazor, <br />
+                Grid for Blazor, <br />
+                Form for Blazor, <br />
+                ValidationMessage for Blazor, <br />
+                ValidationSummary for Blazor, <br />
+                ValidationTooltip for Blazor
+            </td>
+        </tr>
+    </tbody>
+</table>
+
+## Description
+
+This KB answers the following questions:
+
+* How to use conditional required validation with Telerik UI for Blazor components?
+* How to make a Form field required, depending on the value of another field?
+* How to implement a conditional `DataAnnotations` validator and integrate it with the Telerik Blazor Form or Grid?
+* How to display inline validation messages or validation tooltips when using a custom validator?
+
+## Solution
+
+1. Implement a class that inherits from [`ValidationAttribute`](https://learn.microsoft.com/en-us/dotnet/api/system.componentmodel.dataannotations.validationattribute).
+1. Override the [`IsValid()` method overload](https://learn.microsoft.com/en-us/dotnet/api/system.componentmodel.dataannotations.validationattribute.isvalid), which accepts a [`ValidationContext`](https://learn.microsoft.com/en-us/dotnet/api/system.componentmodel.dataannotations.validationcontext) and returns a [`ValidationResult`](https://learn.microsoft.com/en-us/dotnet/api/system.componentmodel.dataannotations.validationresult).
+1. Return a `ValidationResult` that includes the failing field name(s) as a second argument of type `IEnumerable<string>`. This step is crucial in order to apply invalid state to the respective input component and display an inline validation message next to it.
+1. (optional) Override the `FormatErrorMessage` method to provide a custom validation message.
+
+> Creating a custom `DataAnnotations` validator does not involve Telerik APIs and is outside the Telerik support scope. The following implementation is just an example that shows that Telerik Blazor components can work with a custom validator. The exact validator implementation [depends on the specific requirements and can vary](https://stackoverflow.com/questions/26354853/conditionally-required-property-using-data-annotations).
```

Comment:
Switch to [external link syntax](https://docs.telerik.com/style-guide/cross-references#guidelines) for the stack overflow link.

Reply:
```suggestion
> Creating a custom `DataAnnotations` validator does not involve Telerik APIs and is outside the Telerik support scope. The following implementation is just an example that shows that Telerik Blazor components can work with a custom validator. The exact validator implementation <a href="https://stackoverflow.com/questions/26354853/conditionally-required-property-using-data-annotations" target="_blank">depends on the specific requirements and can vary</a>.
```